### DISCUSSION TITLE
Version 2 crashes according to EtreCheck

### DISCUSSION THREAD

**Discussion by TheRoyalScam** [noah-nuebling/mac-mouse-fix#771] [Questions]
I’ve been using MMF for a few weeks, mainly to fix scrolling with my Kensington Expert Mouse Trackball, and it seems to be working nicely…but I ran EtreCheck this morning, as I occasionally do, and it reported 13 crashes of MMF as of this morning. I’ve noticed nothing.

I’m running your latest release of v2 in MacOS Ventura on a MBP i7. Should an upgrade to v3 fix this?

**Comment by TheRoyalScam** [comment]
Bump

**Comment by noah-nuebling** [comment]
Hi there @TheRoyalScam, I'm not sure why these crashes might have happened. If you open Console.app and find the crash reports and send them to me via email, I might be able to figure out more.

Updating to MMF 3 might help, but I don't know since I don't know why the crashes happened exactly.

Let me know if you have any questions.

**Reply by TheRoyalScam** [reply]
Thanks for the reply, Noah. Just emailed two crash reports to your gmail.

**Reply by noah-nuebling** [reply]
Hey there @TheRoyalScam, I looked at your crash reports. It looks like they probably only happen when you detach a device from your computer. In MMF 3, these crashes shouldn't happen, and I also attempted to fix the crashes in MMF 2. You can try my fix here:

[MacMouseFixApp-2.2.4-Alpha.zip](https://github.com/noah-nuebling/mac-mouse-fix/files/13855620/MacMouseFixApp-2.2.4-Alpha.zip)

I hope that helps! Let me know if you have more issues.

**Comment by TheRoyalScam** [comment]
Thanks so much, Noah! I’ll give your v2 fix a shot, and most likely buy v3 even if it’s just to support you.

**Reply by noah-nuebling** [reply]
Great, thank you for for the support! Let me know if you have any more issues or questions.

**Reply by TheRoyalScam** [reply]
Hey Noah, I actually had some similar issues with the Alpha fix. I sent you an email earlier this week about it.

**Reply by noah-nuebling** [reply]
Ah okay sorry, I couldn't find the emails and wasn't sure it was you. I looked at the crash reports, and it seems there is something subtle going on with some objects from Apples IOHID library. I'm not sure if there's some bug in Apples code or if Mac Mouse Fix mishandles those objects somehow. But since these crashes seem to not have any noticable impact on the user experience except for showing up in EtreCheck, and since you're the first person to reach out about them (and I've also never seen them on my system either), I won't prioritize working on these crashes for now. If the crashes still happen on Mac Mouse Fix 3, or if they have an impact on usability of MMF, please let me know though. I might prioritize the problem more highly in that case.

I hope this makes sense. If you have any more thoughts or opinions on this, let me know.

**Reply by noah-nuebling** [reply]
You also mentioned high CPU usage in your email - can you elaborate on that? In what circumstances is the CPU usage high? Did it go up under some specific circumstance? How high is the CPU usage? Anything else that might be relevant for diagnosing this? Thank you!

**Reply by TheRoyalScam** [reply]
Totally makes sense, Noah. As to the High CPU, the .diag file I’d attached should say something about that. I noticed my fans kicking in more often, so ran another EtreCheck and it showed a High CPU alert on Mac Mouse Fix, and pointed to that .diag file.

**Reply by noah-nuebling** [reply]
Oh ok, sorry again, thanks for the heads up. I took a look now. I'm not super familar with these .diag reports, but `90 seconds cpu time over 101 seconds (89% cpu average), exceeding limit of 50% cpu over 180 seconds` seems extremely high. If I understand correctly, that's definitely something, I'd like to address. I tried to analyze the report more deeply, but I don't have the build files for the 2.2.3 release at the moment. (And the .diag file seems to stem from that) If you can reproduce this using the 2.2.4 Alpha I sent you or using Mac Mouse Fix 3, then it would be really great if you could send me the .diag report for that.

If you don't have time, I also understand, just let me know, and I'll try to reproduce the build files for 2.2.3 and see if I can analyze the the report you already sent using that.

**Reply by TheRoyalScam** [reply]
So far I don’t see high CPU warnings in Console for MMF, so I guess the new version fixed that issue. If I see something again, will let you know.

**Reply by noah-nuebling** [reply]
Ok great. Thank you!
